### PR TITLE
fix: Toleration indentation problem

### PR DESCRIPTION
**Pull request by jessicaxiejw**
If you try to add a toleration, it fails with the exception below. It is failing because the Toleration block is in a different level of indentation as `nvidia.com/gpu:NoSchedule`.

```
error converting YAML to JSON: yaml: line 28: did not find expected key
failed to unmarshal manifest /manifests/state-nv-ipam-cni/040-nv-ipam-node.yaml
github.com/Mellanox/network-operator/pkg/render.(*textTemplateRenderer).renderFile
	/workspace/pkg/render/render.go:155
github.com/Mellanox/network-operator/pkg/render.(*textTemplateRenderer).RenderObjects
	/workspace/pkg/render/render.go:81
github.com/Mellanox/network-operator/pkg/state.(*stateNVIPAMCNI).getManifestObjects
	/workspace/pkg/state/state_nv_ipam_cni.go:157
github.com/Mellanox/network-operator/pkg/state.(*stateNVIPAMCNI).Sync
	/workspace/pkg/state/state_nv_ipam_cni.go:99
github.com/Mellanox/network-operator/pkg/state.(*stateManager).SyncState
	/workspace/pkg/state/manager.go:92
github.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).Reconcile
	/workspace/controllers/nicclusterpolicy_controller.go:144
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1598
failed to render objects
github.com/Mellanox/network-operator/pkg/state.(*stateNVIPAMCNI).getManifestObjects
	/workspace/pkg/state/state_nv_ipam_cni.go:160
github.com/Mellanox/network-operator/pkg/state.(*stateNVIPAMCNI).Sync
	/workspace/pkg/state/state_nv_ipam_cni.go:99
github.com/Mellanox/network-operator/pkg/state.(*stateManager).SyncState
	/workspace/pkg/state/manager.go:92
github.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).Reconcile
	/workspace/controllers/nicclusterpolicy_controller.go:144
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1598
failed to create k8s objects from manifest
github.com/Mellanox/network-operator/pkg/state.(*stateNVIPAMCNI).Sync
	/workspace/pkg/state/state_nv_ipam_cni.go:101
github.com/Mellanox/network-operator/pkg/state.(*stateManager).SyncState
	/workspace/pkg/state/manager.go:92
github.com/Mellanox/network-operator/controllers.(*NicClusterPolicyReconciler).Reconcile
	/workspace/controllers/nicclusterpolicy_controller.go:144
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:122
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:323
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:274
sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2
	/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.14.6/pkg/internal/controller/controller.go:235
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1598
```

### CODE REVIEW

**Review by ykulazhenkov — APPROVED** [review]
thx for the fix @jessicaxiejw!

**Review by ykulazhenkov — APPROVED** [review]

**Review by e0ne — CHANGES_REQUESTED** [review]
please, update commit message to:
fix: Toleration indentation problem

**Comment by jessicaxiejw** [comment]
> please, update commit message to: fix: Toleration indentation problem

@e0ne Done

**Comment by jessicaxiejw** [comment]
Could I ask what the release process is like? Is it possible to cut a release after this gets merged?

File: manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml
Comment (killianmuldoon) [inline manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml:44]: Was this change needed in the same way the one for nv-ipam-node was? I'm in the process of adding unit tests to cover this case, and I don't see this function panicking the same way as the NVIPAM one is.

For context my tests are here (still WIP)

**Review by killianmuldoon — COMMENTED** [review]
Thanks for catching this! 

To answer your question about release cadence - we have a schedule driven release for network-operator but we're close to releasing `24.01`. We should have an rc tag including this soon and it should certainly be included in the next release.

File: manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml
Comment (jessicaxiejw) [reply manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml:44]: I think this is the PR? https://github.com/Mellanox/network-operator/pull/814/files

Feel free to close this one then :)

Comment (killianmuldoon) [reply manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml:44]: Thanks for the quick response. I'd definitely prefer to merge your PR - I was just following up with tests. My PR is build on top of yours and it's WIP until this one merges.

**Review by killianmuldoon — APPROVED** [review]
Thanks again for catching this bug!

File: manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml
Comment (jessicaxiejw) [reply manifests/state-nic-feature-discovery/030-nic-feature-discovery-ds.yaml:44]: Thanks for adding tests

**Comment by jessicaxiejw** [comment]
I am still waiting for @e0ne's approval. I am out of the country next week. Feel free to merge this as you see fit.

**Review by adrianchiris — APPROVED** [review]

**Review by e0ne — APPROVED** [review]

**Pull request merged**
